src/command_modules/azure-cli-iot/azure/cli/command_modules/iot/_help.py

```diff
@@ -10,55 +10,133 @@
+    type: command
+    short-summary: Create an Azure IoT Hub.
+    long-summary: See https://azure.microsoft.com/en-us/services/iot-hub/ for an intro to Azure IoT Hub.
 """
 
 helps['iot hub show'] = """
-            type: command
-            short-summary: Show non-security metadata of an IoT Hub.
+    type: command
+    short-summary: Show non-security metadata of an IoT Hub.
 """
 
 helps['iot hub list'] = """
-            type: command
-            short-summary: List IoT Hubs in your subscription or resource group.
-            long-summary: If resource group is provided, IoT Hubs in target resource group will be listed.
-                        Otherwise, IoT Hubs in your subscription will be listed.
+    type: command
+    short-summary: List IoT Hubs in your subscription or resource group.
+    long-summary: If resource group is provided, IoT Hubs in target resource group will be listed.
+                Otherwise, IoT Hubs in your subscription will be listed.
 """
 
 helps['iot hub show-connection-string'] = """
-            type: command
-            short-summary: Show connection string of target Azure IoT Hub.
-            long-summary: If resource group and IoT Hub name are not provided, connection strings for all IoT Hubs in your subscription will be returned.
-                        If only resource group is provided, connection strings for all IoT Hubs in the resource group will be returned.
-                        If both resource group and IoT Hub name are provided, connection string of the IoT Hub will be returned.
+    type: command
+    short-summary: Show connection string of an IoT Hub.
+    long-summary: If resource group and IoT Hub name are not provided, connection strings for all IoT Hubs in your subscription will be returned.
+                If only resource group is provided, connection strings for all IoT Hubs in the resource group will be returned.
+                If both resource group and IoT Hub name are provided, connection string of the IoT Hub will be returned.
+"""
+
+helps['iot hub consumer-group'] = """
+    type: group
+    short-summary: Manage event hub consumer groups of an IoT Hub.
+"""
+
+helps['iot hub consumer-group create'] = """
+    type: command
+    short-summary: Create an event hub consumer group.
+"""
+
+helps['iot hub consumer-group list'] = """
+    type: command
+    short-summary: List all event hub consumer groups.
+"""
+
+helps['iot hub consumer-group show'] = """
+    type: command
+    short-summary: Get an event hub consumer group.
+"""
+
+helps['iot hub consumer-group delete'] = """
+    type: command
+    short-summary: Delete an event hub consumer group.
+"""
+
+helps['iot hub key'] = """
+    type: group
+    short-summary: Manage shared access policies of an IoT Hub.
+"""
+
+helps['iot hub key list'] = """
+    type: command
+    short-summary: List all shared access policies of an IoT Hub.
+"""
+
+helps['iot hub key show'] = """
```

Comment:
I would not recommend you transform the result object. If you have any generic update (which I don't know if you do), the user has to rely on the output of "show" showing the correct object structure and property names.

I'm fine with 'key' based on the reasons you give. I just wanted to verify that that's what your team wants to move forward with.